core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala

```diff
@@ -1415,6 +1415,79 @@ class BlockManagerSuite extends SparkFunSuite with Matchers with BeforeAndAfterE
       super.fetchBlockSync(host, port, execId, blockId)
     }
   }
+
+  def testGetOrElseUpdateForLargeBlock(storageLevel: StorageLevel) {
```

Comment:
@cloud-fan , @vanzin ,
taking the 'parameterized approach', I'd remove most of the tests from `BlockManagerSuite` as they'd require propagating this parameter to too many subsystems.
so, I'm going to modify `DiskStore` and `DiskStoreSuite` to use such a parameter, I'm not sure about leaving a test-case in `BlockManagerSuite` that tests `DISK_ONLY` persistence, what do you guys think?